osu.Game.Rulesets.Osu/Edit/OsuBeatmapVerifier.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit;
+using osu.Game.Rulesets.Edit.Checks.Components;
+using osu.Game.Rulesets.Osu.Edit.Checks;
+
+namespace osu.Game.Rulesets.Osu.Edit
+{
+    public class OsuBeatmapVerifier : BeatmapVerifier
+    {
+        private readonly List<ICheck> checks = new List<ICheck>
+        {
+            new CheckOffscreenObjects()
+        };
+
+        public override IEnumerable<Issue> Run(IBeatmap beatmap)
+        {
+            // Also run mode-invariant checks.
+            foreach (var issue in base.Run(beatmap))
+                yield return issue;
```

Comment:
I think the only situation they would want to silence ruleset-invariant checks is if their ruleset overrides parts of the general Ranking Criteria which none of the 4 base modes do, for example:
- Allowing higher resolution backgrounds than regular.
- Being more lenient with drain time requirements.
- Not requiring maps to be hitsounded.
- Allowing audio files to be >192 kbps.

So if they are allowed to deviate like that, then this would be a case of that. The majority of ruleset-invariant checks would probably be kept, though. So it would end up as something like:
```cs
foreach (var issue in base.Run(beatmap))
{
    if (!(issue.Check is CheckHitsoundVolume || issue.Check is CheckSpreadRequirements))
        yield return issue;
}
```

An alternative would be simply constructing general checks we think may want to be overridden in each ruleset-specific `BeatmapVerifier` instead, and then separating the rest one level higher like you said, but knowing what checks may want to be overridden is difficult to guess.

Reply:
Just to keep things simple let's split the base ones out into their own verifier for now, I think. We can deal with exceptions at a later date if required.